sdk/testing/src/main/java/io/opentelemetry/sdk/testing/metrics/TestMetricData.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.testing.metrics;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.metrics.data.Data;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.resources.Resource;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Immutable representation of all data collected by the {@link
+ * io.opentelemetry.sdk.metrics.data.MetricData} class.
+ */
+@Immutable
+@AutoValue
+public abstract class TestMetricData implements MetricData {
+  public static Builder builder() {
+    return new AutoValue_TestMetricData.Builder()
+        .setResource(Resource.empty())
+        .setName("name")
+        .setInstrumentationScopeInfo(InstrumentationScopeInfo.empty())
+        .setDescription("description")
+        .setUnit("1");
```

Comment:
Let's omit the name, description, and unit from this. They're required fields and calling `build()` without first setting them will result in a `IllegalStateException`. This is fine and better than having arbitrary values IMO.

Its also consistent with the behavior of `TestSpanData`.